cpp/include/raft/distance/detail/distance.cuh

```diff
@@ -247,9 +247,11 @@ struct DistanceImpl<raft::distance::DistanceType::InnerProduct,
            Index_ m,
            Index_ n,
            Index_ k,
+#pragma nv_diag_suppress 177
```

Comment:
will this work for older versions of cuda (like cuda 11.2)? 

I'm wondering if we need to use `__NVCC_DIAG_PRAGMA_SUPPORT__` to figure out if there is support for this compiler flag (sorta like I attempted in https://github.com/rapidsai/raft/pull/941/files).